examples/calibrate_tsqrtP.py

```diff
@@ -0,0 +1,328 @@
+        # {{{ Diagnostic to ensure the frequency was properly identified
+        fl.next("Frequency Domain")
+        fl.plot(d)
+        plt.text(
+            x=0.5,
+            y=0.5,
+            s=rf"$\nu_a={nu_a/1e6:0.2f}$ MHz",
+            transform=plt.gca().transAxes,
+        )
+        assert (0 > nu_a + 0.5 * HH_slicewidth) or (
+            0 < nu_a - 0.5 * HH_slicewidth
+        ), "unfortunately the region I want to filter includes DC -- this is probably not good, and you should pick a different timescale for your scope so this doesn't happen"
+        # }}}
+        # {{{ apply HH frequency filter
+        d["t" : (0, nu_a - 0.5 * HH_slicewidth)] *= 0
+        d["t" : (nu_a + 0.5 * HH_slicewidth, None)] *= 0
+        # }}}
+        d.ift("t")
+        indiv_plots(abs(d), "filtered analytic", "red")
+        # }}}
+        # {{{ set up shape of data to drop the calculated beta values into
+        beta_v_t = d.shape.pop("t").alloc(dtype=np.float64)
+        beta_v_t.copy_axes(d).set_units(r"s√W")
+        # }}}
+        for j in range(len(d["t_pulse"])):
+            s = d["t_pulse", j]
+            int_range = abs(s).contiguous(lambda x: x > 0.05 * s.max())[0]
+            # slightly expand int range to include rising edges
+            int_range[0] -= 5e-6
+            int_range[-1] += 5e-6
+            # {{{ plot the integration range of all pulses prior to integrating.
+            #     Serves as diagnostic to ensure the beta is consistently
+            #     increasing.
+            fl.push_marker()
+            fl.basename = None
+            fl.next("collect filtered analytic")
+            fl.plot(abs(s["t":int_range]), alpha=0.3)
+            fl.pop_marker()
+            # }}}
+            beta_v_t["t_pulse", j] = abs(s["t":int_range]).integrate(
+                "t"
+            ).data.item() / np.sqrt(
+                2
+            )  # t * sqrt(P_rms)
+            # {{{ Can't use indiv_plots because we've already indexed the t_pulse
+            # out and we also want to plot the calculated beta on top
+            if skip_plots is not None and j % skip_plots == 0:
+                switch_to_plot(d, j)
+                fl.plot(
+                    abs(s["t":int_range]),
+                    color="black",
+                    label="integrated slice",
+                )
+                plt.ylabel(r"$\sqrt{P}$ / $\sqrt{W}$")
+                plt.text(
+                    int_range[0] * 1e6 - 1,
+                    0.25,
+                    r"$t_{pulse} \sqrt{P_{tx}} = %f \mathrm{μs} \sqrt{\mathrm{W}}$"
+                    % (beta_v_t["t_pulse", j].item() / 1e-6),
+                )
+            # }}}
+        # {{{ show what we observe -- how does β vary with the programmed pulse length
+        fl.basename = None  # reset so all amplitudes are on same plots
+        beta_v_t.set_plot_color_next() #cycle color list to next color
+        thiscolor = beta_v_t.get_plot_color() # set thiscolor to the above
```

Comment:
need both lines, if I set thiscolor to `beta_v_t.set_plot_color_next()` then everytime this color is called it would cycle the color leading to all lines being a different color